### PR TITLE
[Autotuner] CI Smoke Test - Algorithm & Evaluation mode

### PR DESCRIPTION
**Pull request by luarss**
Target args.algorithm and args.eval.

### CODE REVIEW

File: flow/test/test_helper.sh [hunk 1]
```diff
@@ -63,4 +63,43 @@ if [ ! -z ${MAKE_ISSUE+x} ]; then
...
+  PLATFORM=${PLATFORM^^}  
+
+  #echo "Running Autotuner smoke tune test"
+  #python3 -m unittest tools.AutoTuner.test.smoke_test_tune.${PLATFORM^^}TuneSmokeTest.test_tune
+  
+  #echo "Running Autotuner smoke sweep test"
+  #python3 -m unittest tools.AutoTuner.test.smoke_test_sweep.${PLATFORM^^}SweepSmokeTest.test_sweep
+
+  # echo "Running Autotuner smoke sample & iteration test"
+  # python3 -m unittest tools.AutoTuner.test.smoke_test_sample_iteration.${PLATFORM^^}SampleIterationSmokeTest.test_sample_iteration
```
Comment (vvbandeira) [inline flow/test/test_helper.sh]: Remove code instead of commenting it out.

File: tools/AutoTuner/test/smoke_test_algo_eval.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+#     design = "gcd"
+# 
+#     def test_algo_eval(self):
+#          # Run `make` to get baseline metrics (metadata-base-ok.json)
+#          self.make_base()
+#          for command in self.commands:
+#              print(command)
+#              out = subprocess.run(command, shell=True, check=True)
+#              successful = out.returncode == 0
+#              self.assertTrue(successful)
```
Comment (vvbandeira) [inline tools/AutoTuner/test/smoke_test_algo_eval.py]: Remove code instead of commenting it out.

Comment (luarss) [inline tools/AutoTuner/test/smoke_test_algo_eval.py]:
```suggestion
```

Remove

File: .gitignore [hunk 3]
```diff
@@ -90,3 +90,6 @@ docs/build
 GitGuide.md
 mainREADME.md
 build
+
+# autotuner specific
+metadata-base-at.json
```
Comment (vvbandeira) [inline .gitignore:100]: When and where are these created?

File: flow/designs/sky130hd/gcd/autotuner.json [hunk 4]
```diff
@@ -99,3 +99,4 @@
         "step": 0
     }
 }
+
```
Comment (vvbandeira) [inline flow/designs/sky130hd/gcd/autotuner.json]: Why?

File: flow/test/test_helper.sh [hunk 5]
```diff
@@ -63,4 +63,43 @@ if [ ! -z ${MAKE_ISSUE+x} ]; then
...
+    RUN_AUTOTUNER=1
+    ;;
+esac
+if [ $RUN_AUTOTUNER -eq 1 ]; then
+  case $PLATFORM in
+       "gf180" | "nangate45" | "sky130hd_fakestack" | "sky130hs")
+        RUN_AUTOTUNER=0
+        ;;
+  esac
+fi
```
Comment (vvbandeira) [inline flow/test/test_helper.sh]: See suggestion from other PR.

File: flow/test/test_helper.sh [hunk 6]
```diff
@@ -63,4 +63,43 @@ if [ ! -z ${MAKE_ISSUE+x} ]; then
...
+if [ $RUN_AUTOTUNER -eq 1 ]; then
+  # change directory to ../
+  cd ..
+  echo "Install dependencies in Venv"
+  python3 -m venv .venv
+  source .venv/bin/activate
+  pip install -r tools/AutoTuner/requirements.txt
+
+  PLATFORM=${PLATFORM//-/}
+  PLATFORM=${PLATFORM^^}
```
Comment (vvbandeira) [inline flow/test/test_helper.sh]: See comments from other PR.

File: tools/AutoTuner/test/smoke_test_algo_eval.py [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+os.chdir(src_dir)
+
+class BaseAlgoEvalSmokeTest(unittest.TestCase):
+    platform = ""
+    design = ""
+
+    def setUp(self):
+        self.config = os.path.join(cur_dir, f"../../../flow/designs/{self.platform}/{self.design}/autotuner.json")
+        self.experiment = f"smoke-test-algo-eval-{self.platform}"
+        self.reference = os.path.join(cur_dir, f"../../../flow/designs/{self.platform}/{self.design}/metadata-base-at.json")
```
Comment (vvbandeira) [inline tools/AutoTuner/test/smoke_test_algo_eval.py]:
```suggestion
    def setUp(self):
        design_path = f"../../../flow/designs/{self.platform}/{self.design}"
        self.config = os.path.join(cur_dir, f"{design_path}/autotuner.json")
        self.experiment = f"smoke-test-algo-eval-{self.platform}"
        self.reference = os.path.join(cur_dir, f"{design_path}/metadata-base-at.json")
```

**Review by vvbandeira — APPROVED** [review]

**Pull request merged**